src/utils/errorUtils.ts

```diff
@@ -89,29 +91,19 @@ function getStringMessage(
         errorDetailKeys.includes("non_field_errors") &&
         errorDetailValue.non_field_errors
       ) {
-        message = getStringMessage(
-          errorDetailValue.non_field_errors,
-          keyOptions
-        );
+        message = getStringMessage(errorDetailValue.non_field_errors, {
+          keyOptions: options?.keyOptions
+        });
       } else {
         const defaultErrorKey = errorDetailKeys[0];
 
-        // If error detail is array of objects, it is a nested error, so
-        // generation should continue with the original key options
-        if (isArrayOfObjects(errorDetailValue[defaultErrorKey])) {
-          message = getStringMessage(
-            errorDetailValue[defaultErrorKey],
-            keyOptions
-          );
-        } else {
-          message = getStringMessage(errorDetailValue[defaultErrorKey], {
-            // If not an array of objects, generate final error key using options
-            customKey: getErrorKeyForStringMessageGenerator(
-              defaultErrorKey,
-              keyOptions
-            )
-          });
-        }
+        // Start recursion again with the first key's value
+        // `key` should be sent in case `errorDetailValue[defaultErrorKey]` is this recursion's exit value: string[]
+        // `key` then can be processes according to `keyOptions`
```

Comment:
```suggestion
        // Start recursion again with the first key's value
        // `key` should be sent in case `errorDetailValue[defaultErrorKey]` is the recursion's exit value: string[]
        // `key` then can be processed according to `keyOptions`
```